trustregistry/registry/schema.py

```diff
@@ -0,0 +1,47 @@
+import logging
+from typing import Optional
+from fastapi import APIRouter
+from pydantic import BaseModel, validator
+
+logger = logging.getLogger(__name__)
+
+router = APIRouter(prefix="/registry/schema", tags=["schema"])
+
+
+class SchemaBase(BaseModel):
+    id: str
+
+
+class SchemaUpdate(BaseModel):
+    did: Optional[str] = None
+    name: Optional[str] = None
+    version: Optional[str] = None
+
+
+class SchemaRegister(SchemaUpdate, SchemaBase):
+    @validator("did", always=True)
+    def set_did(cls, value, values):
+        return values["id"].split(":")[0]
+
+    @validator("name", always=True)
+    def set_name(cls, value, values):
+        return values["id"].split(":")[1]
+
+    @validator("version", always=True)
+    def set_version(cls, value, values):
+        return values["id"].split(":")[2]
+
+
+@router.post("")
+async def register_schema(schema: SchemaRegister):
+    return schema.dict()
+
+
+@router.put("/{schema_id}")
+async def update_schema(schema_id: str, schema: SchemaUpdate):
```

Comment:
What is the purpose of the schema object?

Reply:
Again just WIP. Was playing around with validation a little bit.

Reply:
Thnaks though. It's nice to have someone doing code reviews and giving feedback again

Reply:
Sorry for reviewing a WIP PR